Runner.php

```diff
@@ -15,32 +15,65 @@ public function __construct($config, $server)
 
     public function queueDirectory($dir, $prefix = '')
     {
+        if(isset($this->config['include_paths'])) {
+            $this->queueWithIncludePaths();
+        } else {
+            $this->queuePaths($dir, $prefix);
+        }
+
+        $this->server->process_work(false);
+    }
+
+    public function queueWithIncludePaths() {
+        foreach ($this->config['include_paths'] as $f) {
+            if ($f !== '.' and $f !== '..') {
+
+                if (is_dir("/code$f")) {
+                    $this->queuePaths("/code$f", "$f/");
+                    continue;
+                }
+
+                if (isset($this->config['config']['file_extensions'])) {
+                    $this->filterByExtension($f);
+                } else {
+                    $this->server->addwork(array("/code/$f"));
+                }
+            }
+
+        }
+    }
+
+    public function queuePaths($dir, $prefix = '') {
         $dir = rtrim($dir, '\\/');
 
         foreach (scandir($dir) as $f) {
-            if (in_array("$prefix$f", $this->config["exclude_paths"])) {
+            if (!isset($this->config['include_paths']) && in_array("$prefix$f", $this->config["exclude_paths"])) {
```

Comment:
How about passing in the exclusions as an array when you call `queuePaths`? That way the `exclude_paths`-based option could pass in `$this->config["exclude_paths"]`, while the `include-paths`-based option could pass in an empty array.